.changeset/lemon-phones-sort.md

```diff
@@ -2,8 +2,21 @@
 "@vercel/edge-config": major
 ---
 
-- Use stale-while-revalidate during development
-- Don't throw generic errors
-- Make the return values read only
-- Use the `privateEdgeConfig` global when available
-- Export a clone function to support the read only behaviour
+ - **Return values are now read-only to improve in-memory caching**
+   In addition to the read-only change, the `@vercel/edge-config` package now also exports a `clone` function that can be used to make the read-only values modifiable again. Applications that used to modify the return value must now clone the value before doing the required changes.
+
+   ```typescript
+    import { get, clone } from '@vercel/edge-config';
+
+    const myArray = await get('listOfAllowedIPs');
+    // myArray.push('127.0.0.1'); // This won't work and might throw an error
+    const myArrayClone = clone(myArray); // Clones the data to make it modifiable
+    myArrayClone.push('127.0.0.1');
+   ```
+
+ - **Use stale-while-revalidate during development**
+   When `@vercel/edge-config` is used during development, with `NODE_ENV` being set to `development`, the most recent result will be kept in-memory and returned instead of waiting for another network request to complete. In the background, a new request will then be started to ensure the in-memory data gets updated.
```

Comment:
This could be a bit more precise, since we are not keeping the last result in memory, but we are actually fetching the full edge config no matter which key is read and then use that to respond to further reads, while refreshing in the background.